tests/test_all_rules.py

```diff
@@ -212,6 +212,24 @@ def test_index_or_data_view_id_present(self):
             """
             self.fail(fail_msg + '\n'.join(failures))
 
+    def test_note_contains_triage_and_analysis(self):
+        """Ensure the note field contains Triage and analysis content for Elastic rules."""
```

Comment:
> Added a unit test that enforces investigation guides for Elastic prebuilt rules

Is this needed? If we are going to go all in for the generated guides, couldn't we have a weekly/monthly automation that generates guides for new rules and open a PR?

Reply:
My preference is have the complete rule at initial release, so users get full picture, and we do not introduce additional updates. I can understand there might be exceptions for urgent releases.

Reply:
IMO: That adds more complexity when reviewing, potentially distracting reviewers from what matters the most: rule logic

Plus, it adds one more step to get the rules done, as authors would need to fix the generated guide before pushing the PR

Reply:
FWIW, we now have a GitHub action job that can run on a detection-rules branch to generate a guide (given the rule uuid), so the idea would be to generate the guide assuming it follows the same general guidance as this PR.

Reply:
In addition to this https://github.com/elastic/detection-rules/pull/4358#discussion_r1925613069

Any low medium rules without guides will be generated as part of the release prior to shipping.